hack/cleanup.sh

```diff
@@ -10,40 +10,47 @@ select yn in "Yes" "No"; do
 done
 
 # crictl redirect STDOUT.  When no objects (pod, image, container) are present, crictl dump the help menu instead.  This may be confusing to users.
-sudo bash -c '
-    echo "Stopping microshift"
-    set +e
-    systemctl stop --now microshift 2>/dev/null
-    systemctl disable microshift 2>/dev/null
-    systemctl stop --now microshift-containerized 2>/dev/null
-    systemctl disable microshift-containerized 2>/dev/null
-    podman stop microshift 2>/dev/null
-    podman stop microshift-aio 2>/dev/null
-
-    echo "Removing crio pods"
-    until crictl rmp --all --force 1>/dev/null; do sleep 1; done
-
-    echo "Removing crio containers"
-    crictl rm --all --force 1>/dev/null
-
-    echo "Removing crio images"
-    crictl rmi --all --prune 1>/dev/null
-
-    echo "Killing conmon, pause processes"
-    pkill -9 conmon
-    pkill -9 pause
-    pkill -9 ovn-controller
-    pkill -9 ovn-northd
-    pkill -9 ovsdb-server
-
-    echo "Removing /var/lib/microshift"
-    crio wipe -f
-    systemctl restart crio
-    echo "Reverting microshift-ovs-init.service configuration"
-    /usr/bin/configure-ovs.sh OpenShiftSDN
-    rm -rf /var/lib/microshift
-    rm -rf /var/lib/ovn
-    rm -rf /var/run/ovn
-
-    echo "Cleanup succeeded"
-'
+echo "Stopping microshift"
+set +e
+systemctl stop --now microshift 2>/dev/null
+systemctl disable microshift 2>/dev/null
+systemctl stop --now microshift-containerized 2>/dev/null
+systemctl disable microshift-containerized 2>/dev/null
+podman stop microshift 2>/dev/null
+podman stop microshift-aio 2>/dev/null
+
+echo "Removing crio pods in order"
+NON_OVN_PODS=$(crictl pods | grep -vE ovn | grep -v CREATED | awk '{print $1}');
```

Comment:
Instead of "grep -v CREATED" -> tail -n +2

Reply:
I think this expression may be bit more selective. I'm a bit concerned with "ovn" pattern.

crictl pods | tail -n +2 | awk '{if ($7~/openshift-ovn-kubernetes/) next; print}' | awk '{print $1}'